File: plasmid/source/core-plugins/eln-lims/1/as/webapps/eln-lims/html/lib/openbis/js/v3/as/dto/common/search/SearchCriteriaToStringBuilder.js
/**
 * @author pkupczyk
 */
define([ "stjs", "as/dto/common/search/AbstractCompositeSearchCriteria" ], function(stjs, AbstractCompositeSearchCriteria) {
	var SearchCriteriaToStringBuilder = function() {
	};
	stjs.extend(SearchCriteriaToStringBuilder, null, [], function(constructor, prototype) {
		prototype['@type'] = 'as.dto.common.search.SearchCriteriaToStringBuilder';
		prototype.name = null;
		prototype.operator = null;
		prototype.criteria = null;
		prototype.setName = function(name) {
			this.name = name;
			return this;
		};
		prototype.setOperator = function(operator) {
			this.operator = operator;
			return this;
		};
		prototype.setCriteria = function(criteria) {
			this.criteria = criteria;
			return this;
		};
		prototype.toString = function(anIndentation) {
			// TODO rewrite to JS version
			var sb = new StringBuilder();
			var indentation = anIndentation;
			if (indentation.isEmpty()) {
				sb.append(this.name.toUpperCase() + "\n");
			} else {
				sb.append(indentation + "with " + this.name.toLowerCase() + ":\n");
			}
			indentation += "    ";
			if (this.operator != null) {
				sb.append(indentation + "with operator '" + this.operator + "'\n");
			}
			for ( var criterion in this.criteria) {
				if (stjs.isInstanceOf(criterion.constructor, AbstractCompositeSearchCriteria)) {
					var compositeCriteria = criterion;
					sb.append(compositeCriteria.toString(indentation));
				} else {
					sb.append(indentation + criterion.toString() + "\n");
				}
			}
			return sb.toString();
		};
	}, {
		operator : {
			name : "Enum",
			arguments : [ "SearchOperator" ]
		},
		criteria : {
			name : "Collection",
			arguments : [ "ISearchCriteria" ]
		}
	});
	return SearchCriteriaToStringBuilder;
})

File: openbis_api/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/common/search/SearchCriteriaToStringBuilder.java
/*
 * Copyright 2014 ETH Zuerich, Scientific IT Services
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package ch.ethz.sis.openbis.generic.asapi.v3.dto.common.search;

import java.io.Serializable;
import java.util.Collection;

import com.fasterxml.jackson.annotation.JsonIgnoreType;

/**
 * @author pkupczyk
 */
@JsonIgnoreType
public class SearchCriteriaToStringBuilder implements Serializable
{

    private static final long serialVersionUID = 1L;

    private String name;

    private SearchOperator operator;

    private Collection<ISearchCriteria> criteria;

    private boolean negated;

    public SearchCriteriaToStringBuilder setName(String name)
    {
        this.name = name;
        return this;
    }

    public SearchCriteriaToStringBuilder setOperator(SearchOperator operator)
    {
        this.operator = operator;
        return this;
    }

    public SearchCriteriaToStringBuilder setCriteria(Collection<ISearchCriteria> criteria)
    {
        this.criteria = criteria;
        return this;
    }

    public SearchCriteriaToStringBuilder setNegated(final boolean negated)
    {
        this.negated = negated;
        return this;
    }

    public String toString(String anIndentation)
    {
        StringBuilder sb = new StringBuilder();
        String indentation = anIndentation;

        if (negated)
        {
            sb.append("NEGATED ");
        }

        if (indentation.isEmpty())
        {
            sb.append(name.toUpperCase() + "\n");
        } else
        {
            sb.append(indentation + "with " + name.toLowerCase() + ":\n");
        }

        indentation += "    ";

        if (operator != null && criteria.size() > 1)
        {
            sb.append(indentation + "with operator '" + operator + "'\n");
        }

        for (ISearchCriteria aCriteria : criteria)
        {
            if (aCriteria instanceof AbstractCompositeSearchCriteria)
            {
                AbstractCompositeSearchCriteria compositeCriteria = (AbstractCompositeSearchCriteria) aCriteria;
                sb.append(compositeCriteria.toString(indentation));
            } else
            {
                sb.append(indentation + aCriteria.toString() + "\n");
            }
        }
        return sb.toString();
    }

    @Override
    public String toString()
    {
        return toString("");
    }

}
